analysis/src/romgen.rs

```diff
@@ -53,11 +57,26 @@ impl<T: FieldElement> RomGenerator<T> {
             // the rom is exactly the body of the main function!
             let rom = main.body.clone();
 
-            Rom {
+            machine.rom = Some(Rom {
                 statements: rom.statements,
                 batches: None,
-            }
-        });
+            })
+        } else {
+            let latch = latch.expect("static machine should have a latch as parameter");
+
+            // add the necessary embedded constraints which apply to both static and dynamic machines
+            let embedded_constraints = [
+                // inject the function_id
+                parse_pil_statement("pol witness function_id"),
+                // the function id must be constant within a block
+                parse_pil_statement(&format!("(1 - {latch}) * (function_id' - function_id) = 0")),
+            ];
```

Comment:
Yes this will be changed. What are you referring to exactly? One thing I'm thinking of doing is making the `function_id` also user defined so it's not injected. Anything else?

Reply:
I meant the hardcoded PIL code

Reply:
That's just a clearer way to insert this constraint (which is now generic over both the latch and the function_id). It would need to be inserted anyway to have a function abstraction on top of PIL, so in the future it would be inserted "in the language" when turning a machine into PIL.

Actually, we can unify VMs and block machines by turning the former into the latter in romgen,  (with the latch being `instr_return` and the function_id being an internal witness column). Then this constraint `(1 - {latch}) * ({function_id}' - {function_id}) = 0` can be added when compiling them down to PIL.

So:
```
                        user                     
                          |
                          v
user -> VM -> machine(latch, function_id) -> PIL
```